spec/support/shared_examples/transform.rb

```diff
@@ -0,0 +1,808 @@
+shared_examples "a directory transformer" do
+  describe '#into' do
+    subject do
+      DirectoryDiff.transform(source_directory).into(target_directory, options)
+    end
+    let(:options) { {} }
+
+    context 'the current version is an empty directory' do
+      let(:current_directory) { [] }
+      let(:new_directory) do
+        [
+          ['Kamal Mahyuddin', 'kamal@envoy.com', '415-935-3143', nil, 'foo'],
+          ['Matthew Johnston', 'matthew@envoy.com', '415-441-3232', nil],
+          ['Adolfo Builes', 'adolfo@envoy.com', nil, nil]
+        ]
+      end
+
+      it 'returns :insert ops for each row and passes through extra fields untouched' do
+        expect(subject).to eq([
+          [:insert, 'Kamal Mahyuddin', 'kamal@envoy.com', '415-935-3143', nil, 'foo'],
+          [:insert, 'Matthew Johnston', 'matthew@envoy.com', '415-441-3232', nil],
+          [:insert, 'Adolfo Builes', 'adolfo@envoy.com', nil, nil]
+        ])
+      end
+    end
+
+    context 'the new version is an empty directory' do
+      let(:current_directory) do
+        [
+          ['Kamal Mahyuddin', 'kamal@envoy.com', '415-935-3143', nil],
+          ['Matthew Johnston', 'matthew@envoy.com', '415-441-3232', nil]
+        ]
+      end
+      let(:new_directory) { [] }
+
+      it 'returns :delete ops' do
+        expect(subject).to eq([
+          [:delete, 'Kamal Mahyuddin', 'kamal@envoy.com', '415-935-3143', nil],
+          [:delete, 'Matthew Johnston', 'matthew@envoy.com', '415-441-3232', nil]
+        ])
+      end
+    end
+
+    context 'both versions are empty' do
+      let(:current_directory) { [] }
+      let(:new_directory) { [] }
+
+      it 'returns no ops' do
+        expect(subject).to eq([])
+      end
+    end
+
+    context 'the current version is exactly the same as the new directory except for the extra field' do
+      let(:current_directory) do
+        [
+          ['Kamal Mahyuddin', 'kamal@envoy.com', '415-935-3143', nil],
+          ['Matthew Johnston', 'matthew@envoy.com', '415-441-3232', nil]
+        ]
+      end
+      let(:new_directory) do
+        [
+          ['Matthew Johnston', 'matthew@envoy.com', '415-441-3232', nil, 'foo'],
+          ['Kamal Mahyuddin', 'kamal@envoy.com', '415-935-3143', nil]
+        ]
+      end
+
+      it 'returns :noops ops' do # reordered to show order doesn't matter
+        expect(subject).to eq([
+          [:noop, 'Kamal Mahyuddin', 'kamal@envoy.com', '415-935-3143', nil],
+          [:noop, 'Matthew Johnston', 'matthew@envoy.com', '415-441-3232', nil, 'foo']
+        ])
+      end
+    end
+
+    context 'the current version is exactly the same as the new directory' do
+      let(:current_directory) do
+        [
+          ['Kamal Mahyuddin', 'kamal@envoy.com', '415-935-3143', nil],
```

Comment:
@kamal do we try to be smart about phones numbers? for example:

`415-935-3143 === +1415-935-3143`?

Reply:
No, we don't try to normalize it in this process or in specs. This _could_ potentially be a source of `updates` if somewhere in the import process normalizes the phone numbers. However, I don't think we do since it's too brittle to guess the right country code.